### PR TITLE
V2/combined login

### PR DESCRIPTION
**Pull request by JoaquinTrinanes**

### CODE REVIEW

File: prisma/seed.ts [hunk 1]
```diff
@@ -28,6 +29,9 @@ const seedUsers = async (n: number = NUMBER) => {
       const name = faker.name.fullName({ firstName, lastName });
 
       return {
+        role: faker.datatype.boolean()
+          ? UserRole.Requestor
+          : UserRole.Applicant,
```
Comment (Headscracher) [inline prisma/seed.ts:34]: Not sure if we can really just randomize this here, as we will have seeded applicants and requestors which should have the right role and not just something randomized

File: src/pages/auth/signin.tsx [hunk 2]
```diff
@@ -1,41 +1,86 @@
...
+      </DarkButton>
+    </form>
+  );
+};
 
 const SignIn: NextPage = () => {
+  const {
+    query: { user: queryUser },
+    isReady,
+  } = useRouter();
```
Comment (Headscracher) [inline src/pages/auth/signin.tsx]: you already get the router below

File: src/pages/auth/signin.tsx [hunk 3]
```diff
@@ -47,60 +92,80 @@ const SignIn: NextPage = () => {
...
+  useOnce(
+    () => {
+      if (
+        queryUser === UserRole.Applicant ||
+        queryUser === UserRole.Requestor
+      ) {
+        setSelectedRole(queryUser);
+        return;
       }
+      setSelectedRole(UserRole.Applicant);
```
Comment (Headscracher) [inline src/pages/auth/signin.tsx]:
```suggestion
      if (
        queryUser === UserRole.Requestor
      ) {
        setSelectedRole(UserRole.Requestor);
      } else {
        setSelectedRole(UserRole.Applicant);
      }
```
nothing necessary, just a bit easier readable imo

**Review by Headscracher — APPROVED** [review]

File: src/pages/auth/signin.tsx (hunk 3 above)
Comment (JoaquinTrinanes) [reply src/pages/auth/signin.tsx]: I wanted it to be explicit so no one could set it's role as admin, but that way is cleaner 😌

File: prisma/seed.ts (hunk 1 above)
Comment (JoaquinTrinanes) [reply prisma/seed.ts:34]: I just pushed a commit that makes it so the application and request seeders only select valid users ✨

File: .env.example [hunk 4]
```diff
@@ -30,6 +30,7 @@ DATABASE_NAME="vegan_hacktivists"
 DATABASE_USER="vegan_hacktivists"
 DATABASE_PASSWORD="supersecretpassword"
 DATABASE_URL="postgresql://${DATABASE_USER}:${DATABASE_PASSWORD}@localhost:5432/${DATABASE_NAME}?schema=public"
+DATABASE_PORT="5432"
```
Comment (Headscracher) [inline .env.example:34]:
```suggestion
DATABASE_PORT="5432"
EMAIL_SERVER_PORT="2525"
```

File: src/server/context.ts [hunk 5]
```diff
@@ -14,7 +14,11 @@ export const createContextInner = async ({
   req,
   res,
 }: trpcNext.CreateNextContextOptions) => {
-  const session = await unstable_getServerSession(req, res, nextAuthOptions);
+  const session = await unstable_getServerSession(
```
Comment (Headscracher) [inline src/server/context.ts:17]: Looks like we don't have to use unstable here anymore https://next-auth.js.org/configuration/nextjs#getserversession

**Review by Headscracher — APPROVED** [review]

**Pull request merged**